lambdas/odf_to_orpml/odf_to_orpml.py

```diff
@@ -0,0 +1,290 @@
+
+    for s in listoffiles:
+        if s.orig_filename == 'content.xml':
+            bh = zip.read(s.orig_filename)
+            element = ET.XML(bh)
+            ET.indent(element)
+            contentXML = ET.tostring(element, encoding='unicode')
+        elif s.orig_filename == 'meta.xml':
+            bh = zip.read(s.orig_filename)
+            element = ET.XML(bh)
+            ET.indent(element)
+            metadataXML = ET.tostring(element, encoding='unicode')
+
+    return contentXML, metadataXML
+
+
+def extract_odf_metadata(metadataXML: str) -> list:
+    '''Extracts the metadata in the ODF'''
+
+    soup = BeautifulSoup(metadataXML, 'lxml')
+    metadata = soup.find('ns0:meta')
+
+    date = datetime.strptime(
+        getattr(metadata.find('dc:date'), 'text', ''),
+        '%Y-%m-%dT%H:%M:%SZ'
+    )
+    date_formatted = datetime.strftime(date, '%Y-%m-%d')
+
+    odf_meta_tags = [
+        {'name': 'DC.title', 'content': getattr(metadata.find('dc:title'), 'text', '')},
+        {'name': 'DC.subject', 'content': getattr(
+            metadata.find('dc:subject'), 'text', '')},
+        {'name': 'DC.date', 'content': date_formatted},
+        {'name': 'DC.publisher', 'content': getattr(
+            metadata.find('creator'), 'text', '')},
+        {'name': 'DC.format', 'content': 'ODF'}
+    ]
+
+    logger.info('Extracted metadata from ODF')
+
+    return odf_meta_tags
+
+
+def extract_odf_text(contentXML: str) -> str:
+    '''Extracts the body of the text in the ODF'''
+
+    soup = BeautifulSoup(contentXML, 'lxml')
+    pageText = soup.findAll(text=True)
+
+    text = str(' '.join(pageText)).replace('\n', '')
+    logger.info('Extracted text from ODF')
+
+    return clean_text(text=text)
+
+
+def process_orpml(text_body: str, odf_meta_tags: dict, s3_metadata: dict) -> str:
+    '''Builds the ORPML document from the metadata and text extracted from the ODF'''
+
+    orpml = BeautifulSoup(
+        '<!DOCTYPE orpml><orpml><head></head><body></body></orpml>',
+        'html.parser'
+    )
+
+    # Finding the time the object was uploaded
+    date_uploaded = datetime.now()
+    date_uploaded_formatted = date_uploaded.strftime('%Y-%m-%dT%H:%M:%S')
+
+    # Turning the S3 metadata into HTML meta tags
+    regulatory_topics = json.loads(s3_metadata.get('topics'))
+    regulatory_topics_formatted = ', '.join(regulatory_topics)
+
+    s3_meta_tags = [
+        {'name': 'DC.identifier', 'content': s3_metadata['uuid']},
+        {'name': 'DTAC.regulatorId', 'content': s3_metadata['regulator_id']},
+        {'name': 'DTAC.userId', 'content': s3_metadata['user_id']},
+        {'name': 'DC.type', 'content': s3_metadata['document_type']},
+        {'name': 'DTAC.status', 'content': s3_metadata['status']},
+        {'name': 'DTAC.regulatoryTopic', 'content': regulatory_topics_formatted},
+        {'name': 'DTAC.dateSubmitted', 'content': date_uploaded_formatted},
+        {'name': 'DTAC.uri', 'content': s3_metadata['uri']},
```

Comment:
Needs changing from DTAC -> DCAT

Reply:
There's a couple things in each lambda that need changing now so I'll wait for the XSD to be formalised and then ensure it passes validation